frontend/src/Editor/Components/DropDown.jsx

```diff
@@ -16,23 +16,44 @@ export const DropDown = function DropDown({
   registerAction,
   dataCy,
 }) {
-  let { label, value, display_values, values } = properties;
+  let { label, value, advanced, schema, placeholder, display_values, values } = properties;
+
   const { selectedTextColor, borderRadius, visibility, disabledState, justifyContent } = styles;
-  const [currentValue, setCurrentValue] = useState(() => value);
+  const [currentValue, setCurrentValue] = useState(() => (advanced ? findDefaultItem(schema) : value));
+
   const { value: exposedValue } = exposedVariables;
 
-  if (!_.isArray(values)) {
+  function findDefaultItem(schema) {
+    const foundItem = schema?.find((item) => item.default === true);
+    return foundItem ? foundItem.value : null;
+  }
+
+  if (advanced) {
+    values = schema?.map((item) => item.value);
+    display_values = schema?.map((item) => item.label);
+    value = findDefaultItem(schema);
```

Comment:
This is getting called every time when re-render happens. This can be avoided by wrapping with `useRef` and setting it to false after the first time. Then whenever the `schema` changes, this can be recalculated